torch/utils/data/datapipes/utils/common.py

```diff
@@ -16,6 +18,34 @@
 ]
 
 
+def ensure_map_fn_works(fn: Callable, input_col: Optional[Union[int, tuple, list]]):
+    """
+    Checks that function used in a map style datapipe works with the input column
+
+    Args:
+        fn: The function to check.
+        input_col: The input column to check.
+    Returns:
+        None.
+    Raises:
+        TypeError: If the function is not compatible with the input column.
+    """
+    sig = inspect.signature(fn)
+    if isinstance(input_col, (list, tuple)):
+        sz = len(input_col)
+    else:
+        sz = 1
+
+    if len(sig.parameters) >= sz:
+        non_default_params = [p for p in sig.parameters.values() if p.default is p.empty]
+        if len(non_default_params) > sz:
+            raise ValueError(
+                f"The function {fn.__name__} takes {len(non_default_params)} "
+                f"non-default parameters, but {sz} are required."
+            )
+
```

Comment:
super nit: Please keep 2 empty lines between each function.

Reply:
Resolved in 89e4d1e71d7f68f4776972fb73e39df4e2b83391